File: app/src/main/java/es/etologic/mahjongscoring2/domain/model/enums/TableWinds.java
package es.etologic.mahjongscoring2.domain.model.enums;

public enum TableWinds {
    NONE(0),
    EAST(1),
    SOUTH(2),
    WEST(3),
    NORTH(4);

    private int code;

    TableWinds(int code) { this.code = code; }

    public int getCode() { return code; }

    public int getIndex() { return code - 1; }

    public static TableWinds getFromCode(int code) {
        switch (code) {
            default:
                return NONE;
            case 1:
                return EAST;
            case 2:
                return SOUTH;
            case 3:
                return WEST;
            case 4:
                return NORTH;
        }
    }
}

File: app/src/main/java/es/etologic/mahjongscoring2/domain/model/enums/TableWinds.kt
package es.etologic.mahjongscoring2.domain.model.enums

enum class TableWinds(val code: Int) {
    NONE(0),
    EAST(1),
    SOUTH(2),
    WEST(3),
    NORTH(4);
    
    val index: Int
        get() = code - 1
    
    companion object {
        
        fun getFromCode(code: Int): TableWinds {
            return when (code) {
                1 -> EAST
                2 -> SOUTH
                3 -> WEST
                4 -> NORTH
                else -> NONE
            }
        }
    }
}
